easybuild/tools/modules.py

```diff
@@ -555,13 +556,35 @@ def mod_exists_via_show(mod_name):
             # - Check first non-whitespace line for something that looks like an absolute path terminated by a colon
             mod_exists_regex = r'\s*/.+:\s*'
             for line in stderr.split('\n'):
+                # skip whitespace lines
+                self.log.debug("Checking line '%s' to determine whether %s exists...", line, mod_name)
                 if OUTPUT_MATCHES['whitespace'].search(line):
+                    self.log.debug("Treating line '%s' as whitespace, so skipping it", line)
                     continue
+
+                # if any errors occured, conclude that module doesn't exist
                 if OUTPUT_MATCHES['error'].search(line):
+                    self.log.debug("Line '%s' looks like an error, so concluding that %s doesn't exist", mod_name)
                     break
-                if re.match(mod_exists_regex, line):
-                    res = True
+
+                # skip warning lines, which may be produced by modules tool but should not be used
+                # to determine whether a module file exists
+                if line.startswith('WARNING: '):
+                    self.log.debug("Skipping warning line '%s'", line)
+                    continue
+
+                # skip lines that start with 'module-' (like 'module-version'),
+                # see https://github.com/easybuilders/easybuild-framework/issues/3376
+                if line.startswith('module- '):
+                    self.log.debug("Skipping line '%s' since it starts with 'module-version'", line)
```

Comment:
Please update the message and maybe remove the first comment line and move the 2nd (same/similar to log message which is enough)

Reply:
fixed in 03fbb3a, but keeping the comments (since that refers to `module-version`)